backend/lib/richard_burton/validation.ex

```diff
@@ -0,0 +1,65 @@
+defmodule RichardBurton.Validation do
+  @moduledoc """
+  Utilities for data validation
+  """
+
+  alias RichardBurton.Repo
+
+  def validate(module, attrs) do
+    case validate_transaction(module, attrs) do
+      {:error, :ok} -> :ok
```

Comment:
It returns `:ok` because it's valid, there's nothing to add.

I use a transaction to force database validations on the changeset (so I don't have to use `Repo.exists?`, `Repo.get_by` or any function that requires me to know the unique keys here). As this is just for validation, the transaction is always rolled back. `:ok` means the transaction would've succeeded because the changeset is valid.

Reply:
Yes, I agree we should return `:ok` here so maybe I didn't explain myself correctly.

What I find confusing is the fact that `validate_transaction/2` returns `{:error, :ok}` or `{:error, errors`}
Can we just make that function to return either `:ok` or `{:error, error}`?

Reply:
Then it wouldn't make sense to have it in a different function. Because I'd have to:

```elixir
case Repo.transaction(...) do
  {:error, :ok} -> :ok
  {:error, errors} -> {:error, :errors}
end
```

And that's exactly what `validate` does. The point of `validate_transaction` is to keep readability by not having the whole transaction handler inside the case as an argument.